addons/wiki/migrations/0009_migrate_node_wiki_pages.py

```diff
@@ -0,0 +1,195 @@
+    return guid
+
+def create_wiki_page(node, node_wiki, page_name):
+    wp = WikiPage(
+        page_name=page_name,
+        user_id=node_wiki.user_id,
+        node=node,
+        created=node_wiki.date,
+        modified=node_wiki.modified,
+    )
+    wp.update_modified = False
+    return wp
+
+def create_wiki_version(node_wiki, wiki_page):
+    wv = WikiVersion(
+        wiki_page=wiki_page,
+        user_id=node_wiki.user_id,
+        created=node_wiki.date,
+        modified=node_wiki.modified,
+        content=node_wiki.content,
+        identifier=node_wiki.version,
+    )
+    wv.update_modified = False
+    return wv
+
+def create_guids():
+    content_type = ContentType.objects.get_for_model(WikiPage)
+    progress_bar = progressbar.ProgressBar(maxval=WikiPage.objects.count()).start()
+    logger.info('Creating new guids for all WikiPages:')
+    for i, wp in enumerate(WikiPage.objects.all(), 1):
+        # looping instead of bulk_create, so _id's are not the same
+        progress_bar.update(i)
+        Guid.objects.create(object_id=wp.id, content_type_id=content_type.id)
+    progress_bar.finish()
+    logger.info('WikiPage guids created.')
+    return
+
+def create_wiki_pages(nodes):
+    wiki_pages = []
+    progress_bar = progressbar.ProgressBar(maxval=nodes.count()).start()
+    logger.info('Starting migration of WikiPages:')
+    for i, node in enumerate(nodes, 1):
+        progress_bar.update(i)
+        for wiki_key, version_list in node.wiki_pages_versions.iteritems():
+            if version_list:
+                node_wiki = NodeWikiPage.objects.filter(former_guid=version_list[0]).only('user_id', 'date', 'modified').include(None)[0]
+                latest_page_name = NodeWikiPage.objects.filter(former_guid=version_list[-1]).values_list('page_name', flat=True).include(None)[0]
+                wiki_pages.append(create_wiki_page(node, node_wiki, latest_page_name))
+        if i % 500 == 0:
+            with disable_auto_now_add_fields(models=[WikiPage]):
+                WikiPage.objects.bulk_create(wiki_pages, batch_size=1000)
+                wiki_pages = []
+    progress_bar.finish()
+    # Create the remaining wiki pages that weren't created in the loop above
+    with disable_auto_now_add_fields(models=[WikiPage]):
+        WikiPage.objects.bulk_create(wiki_pages, batch_size=1000)
+    logger.info('WikiPages saved.')
+    create_guids()
+    return
+
+def create_wiki_versions(nodes):
+    wp_content_type_id = ContentType.objects.get_for_model(WikiPage).id
+    wiki_versions_pending = []
+    guids_pending = []
+    progress_bar = progressbar.ProgressBar(maxval=nodes.count()).start()
+    logger.info('Starting migration of WikiVersions:')
+    for i, node in enumerate(nodes, 1):
+        progress_bar.update(i)
+        for wiki_key, version_list in node.wiki_pages_versions.iteritems():
+            if version_list:
+                node_wiki = NodeWikiPage.objects.get(former_guid=version_list[0])
```

Comment:
Drive-by: You can do `.filter(...).include(None)[0]` here too

Reply:
Thanks...these changes do not seem to be speeding up the migration on staging data.  Maybe it will be different on prod.